research/management/commands/orcid-researcher-import.py

```diff
@@ -0,0 +1,231 @@
+
+
+    def match_teledata_records(self):
+        """
+        Loop through the found ORCID IDs,
+        gather more information and attempt
+        to match them up with our teledata
+        records.
+        """
+        headers = {
+            'Accept': 'application/json'
+        }
+
+        self.progress_bar = ChargingBar(
+            'Processing',
+            max=len(self.orcid_ids_to_import)
+        )
+
+        for idx, orcid_id in enumerate(self.orcid_ids_to_import):
+            self.progress_bar.next()
+            try:
+                Researcher.objects.get(
+                    orcid_id=orcid_id
+                )
+
+                # Not really, but it makes for better
+                # at-a-glance understanding
+                self.updated += 1
+                # Nothing to do, continue
+                continue
+            except Researcher.DoesNotExist:
+                existing = None
+
+            request_url = "{0}{1}".format(
+                settings.ORCID_BASE_API_URL,
+                orcid_id
+            )
+
+            data = self.__request_records(0, request_url, {}, headers)
+
+            try:
+                first_name = data['person']['name']['given-names']['value']
+                last_name = data['person']['name']['family-name']['value']
+            except KeyError:
+                self.stdout.write("Unable to get first or last name from record.\n")
+                self.error += 1
+                continue
+            except TypeError:
+                self.stdout.write("Unable to get first or last name from record.\n")
+                self.error += 1
+                continue
+
+            try:
+                person = Staff.objects.get(
+                    Q(
+                        first_name__iexact=first_name,
+                        last_name__iexact=last_name
+                    )
+                )
+
+                created = Researcher.objects.create(
+                    orcid_id=orcid_id,
+                    teledata_record=person
+                )
+
+                if created:
+                    self.created += 1
+
+            except Staff.DoesNotExist:
+                continue
+                # self.stderr.write('Cannot find match for {0} {1}\n'.format(first_name, last_name))
+            except Staff.MultipleObjectsReturned:
+                continue
+                # self.stderr.write('Well... whoops... There is more than one person with the name {0} {1}'.format(first_name, last_name))
+
+
+    def __request_records(self, page_num, request_url, params, headers):
+        """
+        Private helper method for retrieving ORCID records
+        """
+        if page_num != 0:
+            params['start'] = page_num * 100
+
+        response = requests.get(request_url, params, headers=headers)
+
+        if response.status_code != response.ok:
+            CommandError("There was an error retriving the results from ORCID")
```

Comment:
`CommandError` needs to be `raise`'d

Reply:
Also, `reponse.ok` returns a boolean value indicating whether or not the response returned an "ok" response. So this if statement was totally wrong. Updated.